e2e/create-cluster_test.go

```diff
@@ -634,4 +634,53 @@ var _ = Describe("CreateCluster", func() {
 		By("Expecting both KubevirtMachines stabilize to a ready=true again.")
 		waitForMachineReadiness(2, 0)
 	})
+
+	// This test will create a tenant cluster from `templates/cluster-template-ext-infra.yaml` template.
+	// 1. generate secret in 'capk-system' namespace, containing external infra kubeconfig and namespace data
+	// (note, the kubeconfig actually points to the same cluster, which is fine for the sake of the test)
+	// 2. generate tenant cluster yaml manifests with `clusterctl generate cluster` command
+	// 3. apply generated tenant cluster yaml manifests
+	// 4. verify that tenant cluster machines booted and bootstrapped
+	// 5. verify that tenant cluster control plane came up successfully
+	It("Create a simple tenant cluster on external infrastructure", func() {
+		By("generating a secret with external infrastructure kubeconfig and namespace")
+		kubeconfig, err := os.ReadFile(os.Getenv("KUBECONFIG"))
+		Expect(err).ToNot(HaveOccurred())
+		externalInfraSecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "external-infra-kubeconfig",
+				Namespace: "capk-system",
```

Comment:
Maybe better to use different namespace, not the one used for the capk controller pod

Reply:
Never mind, my mistake